### ISSUE TITLE
All items in index view are submitted to nova action instead of selected ones.

### ISSUE THREAD

**Issue by timkerr222** [laravel/nova-issues#3586]
- Laravel Version: 8
- Nova Version: 3
- PHP Version: 8
- Database Driver & Version: MySQL
- Operating System and Version: Linux
- Browser type and version: Chrome
- Reproduction Repository: https://github.com/###/###

### Description:

All items in index view are passed to Nova Action instead of only the selected ones. 

### Detailed steps to reproduce the issue on a fresh Nova installation:

https://youtu.be/enCvJzgFOi8

**Comment by timkerr222** [comment]
`<?php

namespace App\Nova\Actions;

use Illuminate\Bus\Queueable;
use Illuminate\Contracts\Queue\ShouldQueue;
use Illuminate\Queue\InteractsWithQueue;
use Illuminate\Support\Collection;
use Laravel\Nova\Actions\Action;
use Laravel\Nova\Fields\ActionFields;

class InvoiceFill extends Action
{
    use InteractsWithQueue, Queueable;

    /**
     * Perform the action on the given models.
     *
     * @param  \Laravel\Nova\Fields\ActionFields  $fields
     * @param  \Illuminate\Support\Collection  $models
     * @return mixed
     */
    public function handle(ActionFields $fields, Collection $models)
    {
        $message = [];
        foreach($models as $model){
            
            try{

                //$fill = \App\Models\Fill::find($model->id);
                //$fill->update([
                  //  "rx_status" => "invoiced"
                //]);
                //$message .= "Fill $model->id update successful!";
                $message[] = ["className" => get_class($model), "id" => $model->id];
            }
            catch(Exception $e){
                $message .= $e->getMessage().PHP_EOL;
            }

            
        }
        return Action::message(json_encode($message));
    }

    /**
     * Get the fields available on the action.
     *
     * @return array
     */
    public function fields()
    {
        return [];
    }
}
`

**Comment by crynobone** [comment]
> one thing that is a little unusual about my use case, is this resource uses a model that is already used by another resource. The resource shares the same model Fill::class.

I need to be able to reproduce the issue before we can apply a fix. So far was unable to reproduce.

https://user-images.githubusercontent.com/172966/136660986-4ba23688-8afd-428a-99bf-b1f30ddb739b.mp4

**Comment by richardkeep** [comment]
I have experienced a similar bug but got a workaround that fixes. I'll open a new issue with more details
